### ISSUE TITLE
Other explanation on why`no_logic_in_create_state`

### ISSUE THREAD

**Issue by rirjkl19** [dart-lang/lints#101]
I'm trying to play around in flutter 3.7 from scratch and it is my first time seeing this rule/linter when I was designing a reusable widget. `no_logic_in_create_state`. I actually have been using this pattern when I was tinkering with the implementation of the `CupertinoDateSelector` because I saw this kind of approach which I found smart and clever so I have used this approach in all of my other codebase. However in the new flutter lint I faced this problem.

I do understand the rule and I know its a bad practice to pass data to the state. However I find it different from what I'm trying to do. If someone from the team could actually recommend me a different or cleaner approach from what I usually have done. I'm always willing to adjust and learn.

Here is my sample code:

``` dart
class ContentWidget extends StatefulWidget {
  final Content content;

  const ContentWidget({super.key, required this.content});

  @override
  State<ContentWidget> createState() {
    switch (content.type) {
      case ContentType.text:
        return _ContentTextWidgetState();
      case ContentType.image:
        return _ContentImageWidgetState();
      case ContentType.other:
        return _ContentStickerWidgetState();
    }
  }
}

class _ContentTextWidgetState extends State<ContentWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class _ContentImageWidgetState extends State<ContentWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class _ContentOtherWidgetState extends State<ContentWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

```

I have been using the condition before in a different approach like putting it inside the state before calling another instance a of stateful widget. But is that the only way to solve it?

I'm referring to this approach:

``` dart

class ContentWidget extends StatelessWidget {
  final Content content;
  const ContentWidget({super.key, required this.content});

  @override
  Widget build(BuildContext context) {
    switch (content.type) {
      case ContentType.text:
        return const ContentTextWidget();
      case ContentType.image:
        return const ContentImageWidget();
      case ContentType.other:
        return const ContentOtherWidget();
    }
  }
}

class ContentTextWidget extends StatefulWidget {
  const ContentTextWidget({super.key});

  @override
  State<ContentTextWidget> createState() => _ContentTextWidgetState();
}

class _ContentTextWidgetState extends State<ContentTextWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class ContentImageWidget extends StatefulWidget {
  const ContentImageWidget({super.key});

  @override
  State<ContentImageWidget> createState() => _ContentImageWidgetState();
}

class _ContentImageWidgetState extends State<ContentImageWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class ContentOtherWidget extends StatefulWidget {
  const ContentOtherWidget({super.key});

  @override
  State<ContentOtherWidget> createState() => _ContentOtherWidgetState();
}

class _ContentOtherWidgetState extends State<ContentOtherWidget> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

```

I just find the second way to be a lot of code for a simple condition. But is there a real benefit? performance or readability?

**Comment by devoncarew** [comment]
Hi @rirjkl19, I don't believe this is the correct repo for your issue. This repo maintains a curated ruelset for dart package development. The lint you're referring to is a flutter specific lint.

Given that you're asking a question - and not reporting a bug - you might re-post your question in one of the community forums, like stack overflow, or other resources listed on https://flutter.dev/community.
